dspy/adapters/chat_adapter.py

```diff
@@ -168,11 +169,19 @@ def parse(self, signature: Type[Signature], completion: str) -> dict[str, Any]:
                 try:
                     fields[k] = parse_value(v, signature.output_fields[k].annotation)
                 except Exception as e:
-                    raise ValueError(
-                        f"Error parsing field {k}: {e}.\n\n\t\tOn attempting to parse the value\n```\n{v}\n```"
+                    raise AdapterParseError(
+                        adapter_name="ChatAdapter",
```

Comment:
Should this be hard-coded like this? Can't it be inferred?

Reply:
Yea it's the equivalent because this code only exists inside `ChatAdapter`.